src/client/commands/submit.rs

```diff
@@ -157,6 +161,25 @@ fn validate_name(name: String) -> anyhow::Result<String, anyhow::Error> {
     }
 }
 
+fn get_validated_path(cwd_opt: Option<String>) -> PathBuf {
+    match cwd_opt {
+        Some(working_dir) => {
+            let working_dir_path = PathBuf::from(&working_dir);
+            match fs::metadata(&working_dir_path) {
+                Ok(cwd) if cwd.is_dir() => working_dir_path,
+                _ => {
+                    warn!(
+                        "{} is not a valid directory on the current node",
+                        working_dir
+                    );
+                    working_dir_path
+                }
+            }
+        }
```

Comment:
(just FYI, you don't have to change it). This could be written a bit shorter with `if let`:
```rust
if let Err(_) = fs::metadata(&working_dir) {
    warn!(
        "{} is not a valid directory on the current node",
        working_dir
    );
}
working_dir.into()
```
It's a shorter alternative if you only really need one branch of a `match`.

Reply:
Thank you :D I'll change it because this is exactly what I've been obsessing over :D